Switch ints that can't be negative to uints

dmathieu opened the pull request:
I am opening this PR as part of the #118 work.

```
Running benchstat
goos: darwin
goarch: arm64
pkg: github.com/elastic/apm-data/codec
                         │ bench-main  │            bench-branch            │
                         │   sec/op    │   sec/op     vs base               │
Encode/format=json-10      22.04µ ± 2%   22.19µ ± 3%       ~ (p=0.271 n=10)
Encode/format=vtproto-10   1.357µ ± 6%   1.359µ ± 2%       ~ (p=0.971 n=10)
Decode/format=json-10      16.23µ ± 7%   16.28µ ± 9%       ~ (p=0.684 n=10)
Decode/format=vtproto-10   6.423µ ± 3%   6.430µ ± 3%       ~ (p=0.912 n=10)
geomean                    7.472µ        7.496µ       +0.31%

                         │  bench-main  │             bench-branch              │
                         │   bytes/op   │   bytes/op    vs base                 │
Encode/format=json-10      6.169Ki ± 0%   6.169Ki ± 0%       ~ (p=1.000 n=10) ¹
Encode/format=vtproto-10   2.702Ki ± 0%   2.702Ki ± 0%       ~ (p=1.000 n=10) ¹
geomean                    4.083Ki        4.083Ki       +0.00%
¹ all samples are equal

pkg: github.com/elastic/apm-data/input/elasticapm/internal/decoder
                   │ bench-main  │         bench-branch          │
                   │   sec/op    │   sec/op     vs base          │
NDStreamDecoder-10   42.51n ± 7%   43.08n ± 7%  ~ (p=0.529 n=10)

pkg: github.com/elastic/apm-data/model/modelpb
                          │ bench-main  │         bench-branch          │
                          │   sec/op    │   sec/op     vs base          │
APMEventToJSON/to-json-10   11.62µ ± 3%   11.66µ ± 6%  ~ (p=0.684 n=10)
```

marclop commented on input/elasticapm/internal/modeldecoder/v2/metricset_test.go:
```diff
@@ -453,8 +453,8 @@ func TestDecodeMetricsetServiceVersion(t *testing.T) {
 
-func repeatInt64(v int64, n int) []int64 {
-	vs := make([]int64, n)
+func repeatInt64(v uint64, n int) []uint64 {
```
```suggestion
func repeatUint64(v uint64, n int) []uint64 {
```

marclop commented on model/modelpb/util_test.go:
```diff
@@ -60,7 +60,7 @@ func uintPtr(i uint32) *uint32 {
 }
 
-func int64Ptr(i int64) *int64 {
+func int64Ptr(i uint64) *uint64 {
```
```suggestion
func uint64Ptr(i uint64) *uint64 {
```

marclop reviewed (approved):
Changes LGTM. Just left small comments to rename the int64Ptr functions to uint64Ptr.

dmathieu commented:
I have changed the code generation to keep the `elem < 0` check, as some values remain `nullable.Int`, due to them being from json, which doesn't know about signed integers.
And added a nolint to that bit of code generation to make staticcheck happy.

axw commented on input/elasticapm/internal/modeldecoder/generator/slice.go:
```diff
@@ -87,6 +87,7 @@ for _, elem := range val.%s{
 func sliceRuleMinVals(w io.Writer, f structField, rule validationRule) error {
 	fmt.Fprintf(w, `
 for _, elem := range val.%s{
+	//lint:ignore SA4003 don't reject the check for uint
```
This is fine for now, but I think we should probably update the generators to remove the redundant check from the generated Go code. To do that I think we would remove the `minVals` tag from the unsigned int field, and infer a minimum value in the generated JSON Schema.

Could you please open an issue to track this?

dmathieu replied:
See https://github.com/elastic/apm-data/issues/131

axw reviewed (approved):
Thanks!

simitt commented:
@dmathieu there are unfortunately a bunch of conflicts now, do you have bandwith today to get these fixed and merge this PR? Otherwise please ensure to hand over.

dmathieu commented:
Ah yes, sorry. I missed the previous notification. I've rebased and will merge now.
